README.md

```diff
@@ -23,13 +27,98 @@ bundle add sincerely
 rails g sincerely:install
 ```
 
-2. Now let's generate and run a migration to create the `notifications` and `notification_templates` tables and the `Notification` and the `NotificationTemplate` model:
+2. You need to generate and run a migration to create the `notifications` and `notification_templates` tables and the `Notification` and the `NotificationTemplate` model:
 ```bash
 rails g sincerely:migration Notification
 
 rails db:migrate
 ```
 
+3. If you want to enable event callbacks you need to run the `events` task to create the `sincerely_delivery_events` table and the `Sincerely::DeliveryEvent` model:
+```bash
+rails g sincerely:events
+
+rails db:migrate
+```
+
+## Configuration
+1. You need to set the notification model generated in the previous step by setting the `notification_model_name` option in the `config/sincerely.yml` file:
+
+```
+delivery_methods:
+  email:
+    class_name: Sincerely::DeliveryMethods::EmailAwsSes
+    options:
+  sms:
+```
+
+2. You need to set the delivery method for each delivery options (email/sms/push) by setting the `delivery_methods` option in the `config/sincerely.yml` file:
+
+```
+notification_model_name: Notification
+```
+
+## Usage
+1. For a detailed description of configuring AWS SES email sending see the following blog post:
+
+2. Once the email delivery method is configured you need to create an email template which is used to generate the email content:
+
+```
+Sincerely::Templates::EmailLiquidTemplate.create(
+  name: 'test',
+  subject: 'template for sending messages',
+  sender: 'john.doe@gmail.com'
+  html_content: 'hi {{name}}',
+  text_content: 'hi {{name}}'
+)
+```
+
+* name: unique id of the template
+* subject: subject of the generated email (can be overwritten in the notification)
+* sender: email address of the sender
+* html_content: html content of the generated email, you can use the liquid syntax to use variables in the content, see [liquid](https://github.com/Shopify/liquid) for details
+* text_content: text content of the generated email, you can use the liquid syntax to use variables in the content, see [liquid](https://github.com/Shopify/liquid) for details
+
+3. You need to create the notification:
+
+```
+Notification.create(
+  recipient: 'jane.doe@gmail.com',
+  notification_type: 'email',
+  template: template,
+  delivery_options: {
+    template_data: {
+      name: 'John Doe'
+    },
+    subject: 'subject'
+  }
+)
+```
+* recipient: recipient of the notification
+* notification_type: email|sms|push
```

Comment:
Probably we want to add an * saying something like `right now the gem supports only AWS SES emails`.